mffpy/tests/test_cached_property.py

```diff
@@ -0,0 +1,64 @@
+"""
+Copyright 2019 Brain Electrophysiology Laboratory Company LLC
+
+Licensed under the ApacheLicense, Version 2.0(the "License");
+you may not use this module except in compliance with the License.
+You may obtain a copy of the License at:
+
+http: // www.apache.org / licenses / LICENSE - 2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ANY KIND, either express or implied.
+"""
+import pytest
+from mffpy.cached_property import cached_property, drop_cache
+import time
+
+sleep_time = 0.4
+
+
+@pytest.fixture
+def dummy():
+    class Dummy:
+        @cached_property
+        def takes_some_time(self):
+            """some documentation code"""
+            time.sleep(sleep_time)
+            return 'unexpected time span'
+    return Dummy()
+
+
+def test_cached_property(dummy):
+    """Test that property is cached"""
+    # Test that doc string is transferred by decorator
+    assert type(dummy).takes_some_time.__doc__ == """some documentation code"""
+    # first time takes 1 second
+    t0 = time.time()
+    err = dummy.takes_some_time
+    dt = time.time()-t0
+    assert dt == pytest.approx(sleep_time, 1e-1), err
```

Comment:
I changed the tolerance for these approximated assert statements from `1e-2` to `1e-1`. This was because these tests would intermittently fail with the lower tolerance (presumably because the time was too variable). This should still be an acceptable tolerance because the two possible times it might take are 0.4 and 0.0, so a tolerance of 0.1 should be enough to distinguish between those.